### PR TITLE
Feature/445 Unit Tests for central-settlement

### PR DESCRIPTION
**Pull request by ggrg**
Restored test/data folder referred as dataAccess in scripts
Installed module **sinon** as devDependency
Changed module **istanbul**: 1.1.0-alpha.1
Added .istanbul.yml

### CODE REVIEW

File: test/unit/domain/settlement/index.test.js~8af0011eba793e5abd3e561c2a35c8dd007f523e [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/*****
```
Comment (rmothilal) [inline test/unit/domain/settlement/index.test.js~8af0011eba793e5abd3e561c2a35c8dd007f523e]: this seems to be a merge conflict backup file. whats with the file name?

File: test/unit/domain/settlement/index.test.js~8af0011eba793e5abd3e561c2a35c8dd007f523e_0 [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/*****
```
Comment (rmothilal) [inline test/unit/domain/settlement/index.test.js~8af0011eba793e5abd3e561c2a35c8dd007f523e_0]: this seems to be a merge conflict backup file. whats with the file name?

File: test/unit/domain/settlement/index.test.js~HEAD [hunk 3]
```diff
@@ -0,0 +1,27 @@
+/*****
```
Comment (rmothilal) [inline test/unit/domain/settlement/index.test.js~HEAD]: this seems to be a merge conflict backup file. whats with the file name?

File: test/unit/domain/settlement/index.test.js~HEAD_0 [hunk 4]
```diff
@@ -0,0 +1,27 @@
+/*****
```
Comment (rmothilal) [inline test/unit/domain/settlement/index.test.js~HEAD_0]: this seems to be a merge conflict backup file. whats with the file name?

File: test/data/settlements/{settlementId}/participants/{participantId}/accounts/{accountId}.js [hunk 5]
```diff
@@ -0,0 +1,191 @@
+/*****
```
Comment (rmothilal) [inline test/data/settlements/{settlementId}/participants/{participantId}/accounts/{accountId}.js]: Why are you are re-adding the mocked tests that you removed on the last PR?

File: test/data/settlements/{settlementId}/participants/{participantId}.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
+/*****
```
Comment (rmothilal) [inline test/data/settlements/{settlementId}/participants/{participantId}.js]: Why are you are re adding the mocked tests that you removed on the last PR?

File: test/data/settlements/{id}.js [hunk 7]
```diff
@@ -0,0 +1,191 @@
+/*****
```
Comment (rmothilal) [inline test/data/settlements/{id}.js]: Why are you are re adding the mocked tests that you removed on the last PR?

File: test/data/settlements.js [hunk 8]
```diff
@@ -0,0 +1,182 @@
+/*****
```
Comment (rmothilal) [inline test/data/settlements.js]: Why are you are re adding the mocked tests that you removed on the last PR?

File: test/data/settlementWindows/{settlementWindowId}.js [hunk 9]
```diff
@@ -0,0 +1,191 @@
+/*****
```
Comment (rmothilal) [inline test/data/settlementWindows/{settlementWindowId}.js]: Why are you are re adding the mocked tests that you removed on the last PR?

File: test/data/settlementWindows/{id}.js [hunk 10]
```diff
@@ -0,0 +1,191 @@
+/*****
```
Comment (rmothilal) [inline test/data/settlementWindows/{id}.js]: Why are you are re adding the mocked tests that you removed on the last PR?

File: test/data/settlementWindows.js [hunk 11]
```diff
@@ -0,0 +1,115 @@
+/*****
```
Comment (rmothilal) [inline test/data/settlementWindows.js]: Why are you are re adding the mocked tests that you removed on the last PR?

File: test/data/mockgen.js [hunk 12]
```diff
@@ -0,0 +1,45 @@
+/*****
```
Comment (rmothilal) [inline test/data/mockgen.js]: Why are you are re adding the mocked tests that you removed on the last PR?

File: test/unit/domain/settlement/index.test.js~8af0011eba793e5abd3e561c2a35c8dd007f523e (hunk 1 above)
Comment (ggrg) [reply test/unit/domain/settlement/index.test.js~8af0011eba793e5abd3e561c2a35c8dd007f523e]: There was a conflict but it was resolved. Such filenames do not exist in the file tree.

File: test/unit/domain/settlement/index.test.js~8af0011eba793e5abd3e561c2a35c8dd007f523e_0 (hunk 2 above)
Comment (ggrg) [reply test/unit/domain/settlement/index.test.js~8af0011eba793e5abd3e561c2a35c8dd007f523e_0]: There was a conflict but it was resolved. Such filenames do not exist in the file tree.

File: test/unit/domain/settlement/index.test.js~HEAD (hunk 3 above)
Comment (ggrg) [reply test/unit/domain/settlement/index.test.js~HEAD]: There was a conflict but it was resolved. Such filenames do not exist in the file tree.

File: test/unit/domain/settlement/index.test.js~HEAD_0 (hunk 4 above)
Comment (ggrg) [reply test/unit/domain/settlement/index.test.js~HEAD_0]: There was a conflict but it was resolved. Such filenames do not exist in the file tree.

File: test/data/settlements/{settlementId}/participants/{participantId}.js (hunk 6 above)
Comment (ggrg) [reply test/data/settlements/{settlementId}/participants/{participantId}.js]: It is required in src/handlers/settlements/{settlementId}/participants/{participantId}.js

File: test/data/settlements/{settlementId}/participants/{participantId}/accounts/{accountId}.js (hunk 5 above)
Comment (ggrg) [reply test/data/settlements/{settlementId}/participants/{participantId}/accounts/{accountId}.js]: It is required in src/handlers/settlements/{settlementId}/participants/{participantId}/accounts/{accountId}.js

File: test/data/mockgen.js (hunk 12 above)
Comment (ggrg) [reply test/data/mockgen.js]: It is required in both: test/data/settlements/{settlementId}/participants/{participantId}.js and test/data/settlements/{settlementId}/participants/{participantId}/accounts/{accountId}.js

**Comment by ggrg** [comment]
All other test/data/**/* will not be returned back, but these two need returned back to allow server start until they're not longer used by the code base:
```
dataAccess.put[`${request.server.app.responseCode}`](request, h, (error, mock) => {
  if (error) reject(error)
  else if (!mock.responses) resolve()
  else if (mock.responses && mock.responses.code) resolve(Boom.boomify(new Error(mock.responses.message), { statusCode: mock.responses.code }))
  else resolve(mock.responses)
})
```

**Pull request merged**